components/core/src/clp_s/ColumnWriter.hpp

```diff
@@ -111,7 +119,9 @@ class ClpStringColumnWriter : public BaseColumnWriter {
     // Methods inherited from BaseColumnWriter
     void add_value(ParsedMessage::variable_t& value, size_t& size) override;
 
-    size_t store(ZstdCompressor& compressor) override;
+    void store(ZstdCompressor& compressor) override;
+
+    size_t get_total_header_size() const override { return sizeof(size_t); }
```

Comment:
Do we need to add this method for other classes?

Reply:
The default implementation that returns 0 works for all of the other column writers. The clp string column writer is the only one that has an extra header to record the size of the encoded variables column.